src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -52,7 +105,10 @@
    (comidi/GET "/environments" request
                (request-handler request))
    (comidi/GET ["/status/" [#".*" :rest]] request
-               (request-handler request))))
+               (request-handler request))
+
+   (comidi/GET ["/environment_classes" [#".*" :rest]] request
+               ((environment-class-handler jruby-service) request))))
```

Comment:
Yeah, `request-handler-service` needs to die, it should not have any logic beyond "I'm going to hand this entire request off blindly to Puppet", so what you've done here is better.

Except... it seems like you could create the handler fn in a `let` block at the top of this routes fn.  This implementation seems like it will keep creating a new handler fn for every request.